src/EnergyPlus/WaterToAirHeatPump.cc

```diff
@@ -242,174 +231,145 @@ namespace WaterToAirHeatPump {
-                                                     RoutineName,            // Routine name
-                                                     CurrentModuleObject,    // Object Type
-                                                     heatPump.Name,          // Object Name
-                                                     cAlphaFields(8));       // Field Name
-
-                if (!ErrorsFound) {
-                    //     Test PLF curve minimum and maximum. Cap if less than 0.7 or greater than 1.0.
+                case CompressorType::Rotary: {
+                    heatPump.CompPistonDisp = s_ip->getRealFieldValue(fields, schemaProps, "compressor_piston_displacement");
+                    heatPump.CompSucPressDrop = s_ip->getRealFieldValue(fields, schemaProps, "compressor_suction_discharge_pressure_drop");
+                    break;
+                }
+                case CompressorType::Scroll: {
+                    heatPump.RefVolFlowRate = s_ip->getRealFieldValue(fields, schemaProps, "refrigerant_volume_flow_rate");
+                    heatPump.VolumeRatio = s_ip->getRealFieldValue(fields, schemaProps, "volume_ratio");
+                    heatPump.LeakRateCoeff = s_ip->getRealFieldValue(fields, schemaProps, "leak_rate_coefficient");
+                    break;
+                }
+                default: {
+                    ShowSevereError(
+                        state,
+                        format("{}Invalid {} ({}) entered. {}={}", RoutineName, "Compressor Type", compType, CurrentModuleObject, heatPump.Name));
+                    ErrorsFound = true;
+                    break;
+                }
+                }
+                heatPump.SourceSideUACoeff = s_ip->getRealFieldValue(fields, schemaProps, "source_side_heat_transfer_coefficient");
+                heatPump.SourceSideHTR1 = s_ip->getRealFieldValue(fields, schemaProps, "source_side_heat_transfer_resistance1");
+                heatPump.SourceSideHTR2 = s_ip->getRealFieldValue(fields, schemaProps, "source_side_heat_transfer_resistance2");
+                cFieldName = "Part Load Fraction Correlation Curve Name";
+                std::string const coolPLFCurveName = s_ip->getAlphaFieldValue(fields, schemaProps, "part_load_fraction_correlation_curve_name");
+                if (coolPLFCurveName.empty()) {
+                    ShowWarningEmptyField(state, eoh, cFieldName, "Required field is blank.");
+                    ErrorsFound = true;
+                } else if ((heatPump.PLFCurveIndex = Curve::GetCurveIndex(state, coolPLFCurveName)) == 0) {
+                    ShowSevereItemNotFound(state, eoh, cFieldName, coolPLFCurveName);
+                    ErrorsFound = true;
+                } else if (Curve::CheckCurveDims(state, heatPump.PLFCurveIndex, {1}, RoutineName, CurrentModuleObject, heatPump.Name, cFieldName)) {
+                    ShowSevereCustomField(state, eoh, cFieldName, coolPLFCurveName, "Illegal curve dimension.");
```

Comment:
If things don't fit right or are too cumbersome, you can leave the current free-form error messaging in there if you want. 99% of my usage is `ItemNotFound`, `EmptyField`, `DuplicateName`, and `InvalidKey`.

As far a matching the current error lingo exactly ... I don't know if that's an actual goal. The current wording differs from context to context, sometimes quite substantially, depending on the author and/or which module they copied the code from.

Reply:
At this point, I would not spend more time on this. This can be a later pass. Or not.

Reply:
OK. I will make one final pass to address the most apparent changes needed.